kzg_prover/src/circuits/utils.rs

```diff
@@ -198,6 +199,35 @@ pub fn open_user_points(
     )
 }
 
+/// Calculate all opening proofs at once for the polynomials using the amortized KZG approach
+///
+/// # Arguments
+///
+/// * `advice_polys` - the advice polynomials
+/// * `params` - the KZG parameters
+/// * `column_range` - the range of the advice columns of interest
+/// * `omega` - $\omega$, the generator of the multiplicative subgroup used to interpolate the polynomials.
+///
+/// # Returns
+///
+/// * `Vec<Vec<G1>>` - all KZG opening proofs for the polynomials in the range
+pub fn open_user_points_amortized(
+    advice_polys: &[Polynomial<Fp, Coeff>],
+    params: &ParamsKZG<Bn256>,
+    column_range: Range<usize>,
+    omega: Fp,
+) -> Vec<Vec<G1>> {
+    advice_polys[column_range]
+        // Parallelize the independent amortized openings of the user ID and balance polynomials
+        .par_iter()
```

Comment:
I just curious that rayon helps to improve performance in here. because it already `best_fft` well parallelized inside this logic.

Reply:
Rayon is useful here because we need to calculate multiple openings - one opening for the user ID polynomial and a number of openings for the user balance polynomials, depending on the number of currencies.